File: NeetCode150/GroupAnagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        groups = {}
        for x in strs:
            s = ''.join(sorted(x))
            if s in groups:
                groups[s].append(x)
                continue
            groups[s] = [x]
        
        return groups.values()

File: NeetCode150/Array&Hashing/GroupAnagrams.cpp
#include <unordered_map>

class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<string, vector<string>> groups;

        for (auto &s : strs) {
            string copy = s;
            sort(copy.begin(), copy.end());
            groups[copy].push_back(s);

            // not needed c++ automatically creates a default vector
            // if (groups.find(copy) != groups.end()) {
            //     groups[copy].push_back(s);
            //     continue;
            // }

            // groups[copy] = {s};
        }
        vector<vector<string>> ret;
        for (const auto &p: groups) {
            ret.push_back(p.second);
        }
        return ret;
    }
};
